api/src/wfl/executor.clj

```diff
@@ -184,13 +203,16 @@
         current (firecloud/method-configuration workspace methodConfiguration)
         _       (error-on-method-config-version-mismatch
                  current methodConfigurationVersion)
-        inc'd   (inc (:methodConfigVersion current))]
-    (-> "%s Updating %s in %s with {:dataReferenceName %s :methodConfigVersion %s}"
-        (format (log-prefix executor) methodConfiguration workspace name inc'd)
-        log/debug)
-    (firecloud/update-method-configuration
-     workspace methodConfiguration
-     (assoc current :dataReferenceName name :methodConfigVersion inc'd))
+        inc'd   (inc (:methodConfigVersion current))
+        newMc   (merge
+                 current
+                 {:dataReferenceName name :methodConfigVersion inc'd}
+                 (when-let [table (table-from-snapshot-reference executor reference)]
```

Comment:
The `when-let` keeps us from accidentally overwriting an existing table with a `nil`, in the case we don't find a single table match in the snapshot.